.circleci/config.yml

```diff
@@ -0,0 +1,127 @@
+      - run:
+          command: docker build -t app:build .
+          no_output_timeout: 20m
+
+  test:
+    docker:
+      - image: circleci/node:6.14.3-stretch
+      - image: redis
+      - image: circleci/postgres:10-alpine-postgis
+    working_directory: ~/mozilla/redash
+    steps:
+      - checkout
+      - run: mkdir -p /tmp/test-reports/pytest
+      - run: sudo apt-get update
+      - run: sudo apt-get install -y python-pip python-dev
+      - run: sudo apt-get install -y redis-tools redis-server
+      - run: sudo pip install --upgrade setuptools
+      - run: sudo pip install -r requirements_dev.txt
+      - run: sudo pip install -r requirements.txt
+      - run: sudo npm install
+      - run: sudo npm run build
+      - run: 
+          command: pytest --junitxml=/tmp/test-reports/pytest/junit.xml tests/
+          environment:
+            REDASH_REDIS_URL: redis://localhost:6379/0
+            REDASH_DATABASE_URL: "postgresql://postgres@localhost/postgres"
+      - store_artifacts: 
+          path: /tmp/test-reports/
+          destination: tr1
+      - store_test_results:
+          path: /tmp/test-reports/
+
+  deploy:
+    docker:
+      - image: docker:18.02.0-ce
+    working_directory: ~/mozilla/redash
+    steps:
+      - checkout
+      - setup_remote_docker
+      - run:
+          name: Deploy to Dockerhub
+          no_output_timeout: 20m
+          command: |
+            # Deploy master
+            if [ "${CIRCLE_BRANCH}" == "master" ]; then
+              ./bin/deploy "master"
+            else
+            # Deploy a release tag...
+              ./bin/deploy "rc"
+            fi
+
+  deploy-milestone:
+    docker:
+      - image: docker:18.02.0-ce
+    working_directory: ~/mozilla/redash
+    steps:
+      - checkout
+      - setup_remote_docker
+      - run:
+          name: Deploy milestone to Dockerhub
+          no_output_timeout: 20m
+          command: |
+            ./bin/deploy "$CIRCLE_TAG"
+            ./bin/alias "$CIRCLE_TAG" "latest"
+
+
+workflows:
+  version: 2
+  build-test-deploy:
+    jobs:
+      - build:
+          filters:
+            tags:
+              only: /.*/
+            branches:
+              ignore:
+                - gh-pages
+
+      - test:
+          filters:
+            tags:
+              only: /.*/
+            branches:
+              ignore:
+                - gh-pages
+
+      - deploy:
+          requires:
+            - test
+          filters:
+            tags:
+              only: /.*/
+            branches:
+              only:
+                - master
+                - release
+
+      - deploy-milestone:
+          requires:
+            - test
+          filters:
+            tags:
+              only: /^m[0-9]+(\.[0-9]+)?$/
+            branches:
+              only:
+                # Do we only care about milestone tags on these branches?
```

Comment:
i'm not sure what branches we care about deploying milestone tags?

Reply:
Tags are just pointers to a commit and have nothing to do with branches. I think the branches are supposed to be set to ignore based on my reading [here](https://circleci.com/docs/2.0/workflows/#executing-workflows-for-a-git-tag).